reference/promise-types/files.markdown

```diff
@@ -2749,6 +2749,45 @@ separator, since the backward slash has a special meaning in a regular
 expression. Literal paths may also use backslash (`\`) as a path
 separator.
 
+### fsattrs
+
+**Description:** Manage file system attributes
+
+**Type** `body fsattrs`
+
+**See also:** [Common body attributes][Promise types#Common body attributes]
+
+**Notes:**
+File system attributes are not supported on all file systems and platforms.
+Hence, CFEngine will do it's best effort without any guarantees.
+
+#### immutable
+
+**Description:** Set / clear immutable file system attribute
+
+**Type:** [`boolean`][boolean]
+
+**Example:**
+
+```cf3
+body fsattrs set_immutable
+{
+  immutable => "true";
+}
+```
+
+**Notes:**
+Currently only regular files are supported when configuring the immutable bit.
+Furthermore, this will only work on Linux or BSD based systems with filesystems
+that support it. CFEngine will simply ignore the immutable constrain if it is
+not supported.
+
+If the immutable constraint is:
+- set to `"true"`, CFEngine will try to clear the immutable bit (if set) before
+  executing the promise and (re-)set it afterwards.
```

Comment:
But, we won't for example make it false then determine no changes are needed and set it back right? Well only remove that immutable bit temporarily once we know a change is needed.

Reply:
This will be quite difficult to achieve. But I can look into it.

Reply:
:muscle:

Reply:
Now it should only temporarily clear the immutable bit when needed. And the agent will try to keep the period it's cleared as short as machinely possible